backend/amd64/proc.ml

```diff
@@ -133,18 +143,27 @@ let hard_float_reg =
   for i = 0 to 15 do v.(i) <- Reg.at_location Float (Reg (100 + i)) done;
   v
 
+let hard_vec128_reg =
+  let v = Array.make 16 Reg.dummy in
+  for i = 0 to 15 do v.(i) <- Reg.at_location Vec128 (Reg (100 + i)) done;
+  v
+
+(* We don't need to include vec128 regs here, as they use the same IDs as floats. *)
 let all_phys_regs =
   Array.append hard_int_reg hard_float_reg
```

Comment:
I am afraid we need to include it because of the way CSE uses the result of `destroyed_at_oper` as a key to `num_reg` which is of type Reg.Map in which the order is based on stamps.

Reply:
Yeah, just ran into this - restored the destroyed_at behavior including both float and vec128